lib/data_processor.rb

```diff
@@ -0,0 +1,66 @@
+require 'open-uri'
+require 'csv'
+
+class DataProcessor
+
+  def initialize(user_x, user_y, url)
+    @user_x = Float(user_x)
+    @user_y = Float(user_y)
+    @url = URI.parse(url)
+
+  rescue => ex
+    puts "Invalid arguments: #{ex.message}"
+  end
+
+
+  def process_data
+    raise '`user_x` must be a valid coordinate' unless @user_x.between?(-90, 90)
+    raise '`user_y` must be a valid coordinate' unless @user_y.between?(-180, 180)
+
+
+    # read file data
+    file_obj = @url.open { |f| f.read }
+    shop_data = CSV.parse(file_obj)
+
+
+    # get closest shops
```

Comment:
Why did you feel the need to use comments?

Reply:
Not needed if I use single responsibilities